pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -111,6 +111,24 @@ func New(client kube.Client, revision, domainSuffix string) (model.ConfigStoreCa
 	return NewForSchemas(context.Background(), client, revision, domainSuffix, schemas)
 }
 
+var crdWatches = map[config.GroupVersionKind]chan struct{}{
+	gvk.KubernetesGateway: make(chan struct{}),
+}
+
+func WaitForCRD(k config.GroupVersionKind, stop <-chan struct{}) bool {
```

Comment:
Some comment would be nice. Is this a new way to enable features only when the CRD is created ? Wouldn't a callback be better ( i.e. a map of CRD types to init functions ? )

Reply:
Yes, added some comments.

A callback seems nice but I think it may add some complications around the startup logic that makes it a bit more complex. Right now the startup handles releasing leader lock on shutdown, etc that would be a bit harder with callback. If there are more use cases  that callback would be better we can always add it later I think?